langwatch/src/features/onboarding/components/sections/model-provider/ModelProviderConfigFields.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useMemo, useState, useCallback, useEffect } from "react";
+import {
+  VStack,
+  HStack,
+  Button,
+  Text,
+  Field,
+  Spinner,
+} from "@chakra-ui/react";
+import { z } from "zod";
+import { Switch } from "../../../../../components/ui/switch";
+import { getModelProvider } from "../../../regions/model-providers/registry";
+import type { ModelProviderKey } from "../../../regions/model-providers/types";
+import { useOrganizationTeamProject } from "../../../../../hooks/useOrganizationTeamProject";
+import { useModelProvidersSettings } from "../../../../../hooks/useModelProvidersSettings";
+import { useModelProviderForm } from "../../../../../hooks/useModelProviderForm";
+import { useModelProviderFields } from "../../../../../hooks/useModelProviderFields";
+import { DocsLinks } from "../observability/DocsLinks";
+import { modelProviders as modelProvidersRegistry } from "../../../../../server/modelProviders/registry";
+import {
+  parseZodFieldErrors,
+  type ZodErrorStructure,
+} from "../../../../../utils/zod";
+import { ModelProviderCredentialFields } from "./ModelProviderCredentialFields";
+import { ModelProviderExtraHeaders } from "./ModelProviderExtraHeaders";
+import { ModelProviderModelSettings } from "./ModelProviderModelSettings";
+
+interface ModelProviderConfigFieldsProps {
+  modelProviderKey: ModelProviderKey;
+}
+
+const OPENAI_DEFAULT_BASE_URL = "https://api.openai.com/v1";
+
+export const ModelProviderConfigFields: React.FC<ModelProviderConfigFieldsProps> = ({
+  modelProviderKey,
+}) => {
+  const meta = getModelProvider(modelProviderKey);
+  const { project } = useOrganizationTeamProject();
+  const projectId = project?.id;
+
+  const backendKey = meta?.backendKey;
+  const { providers, isLoading, refetch } = useModelProvidersSettings({
+    projectId,
+  });
+
+  const provider = useMemo(() => {
+    if (!backendKey) return void 0;
+
+    const existing = providers?.[backendKey as keyof typeof providers];
+    if (existing) return existing;
+
+    return {
+      provider: backendKey,
+      enabled: false,
+      customKeys: null,
+      models: null,
+      embeddingsModels: null,
+      disabledByDefault: true,
+      extraHeaders: [],
+    } as any;
+  }, [backendKey, providers]);
+
+  const [state, actions] = useModelProviderForm({
+    provider,
+    projectId,
+    projectDefaultModel: meta?.defaultModel ?? project?.defaultModel ?? null,
```

Comment:
why do you need to pass this into a custom hook? It can get it itself and we can avoid prop-drilling. (these aren't even props, they come from cache-optimized hooks)

Reply:
this is an optional field to set the default model when it saves the model provider.